framework/tlvf/tlvf.py

```diff
@@ -1094,50 +1094,52 @@ def addClassVarLenMethods(self, obj_meta, param_type, param_name, param_meta, pa
-    def addAllocationMarkersCreate(self, obj_meta, func_name, param_meta, param_length, memmove):
+    def addAllocationMarkersCreate(self, obj_meta, param_meta, param_length, memmove):
         lines_cpp = []
-        if param_meta.length_type == MetaData.LENGTH_TYPE_VAR:
+        if param_meta.length_type == MetaData.LENGTH_TYPE_VAR or param_meta.length_type == MetaData.LENGTH_TYPE_DYNAMIC:
             if memmove:
-                lines_cpp.append("%suint8_t *src = (uint8_t*) m_%s;" % (self.getIndentation(1), param_meta.name))
+                lines_cpp.append("%suint8_t *src = (uint8_t *)m_%s;" % (self.getIndentation(1), param_meta.name))
                 lines_cpp.append("%sif (!m_parse__) {" %self.getIndentation(1))
+                # Since list of classes does not have length parameter like list of native type, 
+                # taking the buffer pointer from classes vector last element  
                 if param_length != None:
-                    lines_cpp.append("%sif (*m_%s != 0) {" % (self.getIndentation(2), param_length))
-                    ptr = "(uint8_t *)m_%s_vector[*m_%s - 1]->getBuffPtr()" %(param_meta.name,param_length)
+                    lines_cpp.append("%sif (m_%s_idx__ != 0) {" % (self.getIndentation(2), param_meta.name))
+                    ptr = "(uint8_t *)m_%s_vector[m_%s_idx__ - 1]->getBuffPtr()" %(param_meta.name,param_meta.name)
                     lines_cpp.append("%ssrc = %s;" %(self.getIndentation(3), ptr))                    
                     lines_cpp.append("%s}" %self.getIndentation(2))
                 lines_cpp.append("%suint8_t *dst = src + len;" %(self.getIndentation(2)))
                 lines_cpp.append("%ssize_t move_length = getBuffRemainingBytes(src) - len;" %self.getIndentation(2))
                 lines_cpp.append("%sstd::copy_n(src, move_length, dst);" %self.getIndentation(2))
                 lines_cpp.append("%s}" %self.getIndentation(1))
 
-            marker = "%s_%s_%s_%s" %(self.CODE_CLASS_ALLOC_INSERT, obj_meta.name, func_name, param_meta.name)
-            obj_meta.alloc_list.append((marker, func_name))
+            marker = "%s_%s_%s_%s" %(self.CODE_CLASS_ALLOC_INSERT, obj_meta.name, "create", param_meta.name)
+            obj_meta.alloc_list.append((marker, "create"))
             lines_cpp.append(marker)
 
         return lines_cpp
 
-    def addAllocationMarkersAlloc(self, obj_meta, func_name, param_meta, param_length, memmove):
+    def addAllocationMarkersAlloc(self, obj_meta, param_meta, param_length, memmove):
         lines_cpp = []
         if param_meta.length_type == MetaData.LENGTH_TYPE_VAR:
             if memmove:
-                lines_cpp.append("%suint8_t *src = (uint8_t*) &m_%s[*m_%s];" % (self.getIndentation(1), param_meta.name, param_length))
+                lines_cpp.append("%suint8_t *src = (uint8_t *)&m_%s[*m_%s];" % (self.getIndentation(1), param_meta.name, param_length))
```

Comment:
same

Reply:
space should be between the type to the asterisks.